### PR TITLE
Tidy historic rates

### PR DESCRIPTION
**Pull request by KludgeKML**
Historic rate information is not used for a lot of smart answers, so tidy up by removing it and updating any tests that use historic data when the smart answer itself does not.

https://trello.com/c/1jbnKNIK/1531-remove-legacy-data-from-smart-answers-to-make-uprating-easier

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Comment by KludgeKML** [comment]
Not touching the Child Benefit Tax Calculator just yet, as this from Richard Skipper:

"HMRC has told us they're currently signing off a request to remove all the years before 2017/18, so please don't remove anything yet."

**Comment by KludgeKML** [comment]
# Rates Files Changed: 
- minimum_wage.yml: Uses Current and previous year data. Tests maybe require actual numbers  because the calculator does more complex calculations, obsolete historical tests removed. Remaining tests updated, note added that test dates should be updated in YAML.
- marriage_abroad_consular_fees.yml: Historic rates not used, not in any tests
- state_pension.yml: Historic rates not used in UI or any tests
- births_and_deaths_document_return_fees.yml: Historic rates not used except in tests, replaced with non-numeric tests
- register_a_birth.yml: Historic rates not used except in tests, replaced with non-numeric tests
- register_a_death.yml: Historic rates not used except in tests, replaced with non-numeric tests
- married_couples_allowance.yml: Historic rates not used except in tests, simplify tests to only last date. Possible we can simplify allowance logic?
- redundancy_pay.yml: Historic rates over previous 4 tax years used, replaced tests with non-numeric tests to confirm data availability
- redundancy_pay_northern_ireland.yml: Historic rates over previous 4 tax years used, replaced tests with non-numeric tests to confirm data availability

# Rates Files Unchanged
- personal_allowance.yml: No historic data
- statutory_sick_pay.yml: Historic data back to 2010, but UI can use it.
- maternity_paternity_adoption.yml: Historic data back to 2010, but UI can use it.
- maternity_paternity_birth.yml: Historic data back to 2010, but UI can use it.
- child_benefit_rates.yml: Historic data used in UI, HMRC reviewing, leave for now

File: lib/smart_answer/calculators/married_couples_allowance_calculator.rb [hunk 1]
```diff
@@ -35,24 +35,13 @@ def calculate_allowance
...
-        else
-          mca_entitlement = maximum_mca
-        end
+        reduction = (income - income_limit_for_personal_allowances) / 2
+        reduced_mca = maximum_mca - reduction
+        mca_entitlement = if reduced_mca > minimum_mca
+                            reduced_mca
+                          else
+                            minimum_mca
+                          end
```
Comment (1pretz1) [inline lib/smart_answer/calculators/married_couples_allowance_calculator.rb:44]: Nice 👍

**Review by 1pretz1 — APPROVED** [review]
Nice work Keith 👍 Reduced a massive amount of deadwood and code complexity! I also appreciate the YML comments which should make it easier to manage, without having to understand the calculator. 

I wonder whether we could also add comments to the YML files that only have one set of data (current). This is implied as there is only one entry but we could leave a comment saying a max of two entries? One for the current rate and one for the future rate so we can merge changes ahead of the actual live date.

**Comment by KludgeKML** [comment]
Good idea, I'll go through and make sure all the YAMLs have some form of top comment. I'm mildly reluctant to update the end date on register_a_birth etc just because it gives a sort of a clue about how long it's been since the yml had some attention / the actual rates changed. I definitely think when I added in new elements I'd put a hundred-year or whatever end date on them myself, so... Perhaps better to make them all consistent. I'll tweak and re-upload.

File: test/unit/calculators/register_a_birth_rates_query_test.rb [hunk 2]
```diff
@@ -6,31 +6,20 @@ class RegisterABirthRatesQueryTest < ActiveSupport::TestCase
       @rates_query = RatesQuery.from_file("register_a_birth")
     end
 
-    context "for 2015/16" do
+    context "The current and last dates in the file" do
       setup do
-        @rates = @rates_query.rates(Date.parse("2015-04-06"))
+        @rate_array = [
+          @rates_query.rates,
+          @rates_query.rates(Date.parse("2099-01-01")),
```
Comment (edwardkerry) [inline test/unit/calculators/register_a_birth_rates_query_test.rb:13]: should this be `2999`?

Comment (KludgeKML) [reply test/unit/calculators/register_a_birth_rates_query_test.rb]: It should be! Thanks. (and the equivalent _deaths one).

**Pull request merged**